crates/template_lib/src/resource/builder/confidential.rs

```diff
@@ -59,26 +110,72 @@ impl ConfidentialResourceBuilder {
 
     /// Specify a view key for the confidential resource. This allows anyone with the secret key to uncover the balance
     /// of commitments generated for the resource.
-    /// NOTE: it is not currently possible to change the view key after the resource is created.
+    /// 
+    /// Developers can pass a view key of the [`RistrettoPublicKeyBytes`] type.
+    /// 
+    /// # Note
+    /// It is not currently possible to change the view key after the resource is created.
     pub fn with_view_key(mut self, view_key: RistrettoPublicKeyBytes) -> Self {
         self.view_key = Some(view_key);
         self
     }
 
     /// Sets up who can mint new tokens of the resource
+    ///
+    /// Allows you to pass an [`AccessRule`] that defines who can mint new tokens of the resource.
+    ///
+    /// By default, minting is disabled for all users.
+    ///
+    /// #Examples
+    ///
+    /// ```rust, ignore
+    /// use tari_template_lib::auth::AccessRule;
+    /// use tari_template_lib::resource::builder::ResourceBuilder;
+    /// ResourceBuilder::confidential()
+    ///     .mintable(rule!(allow_all))
+    ///     .build();
+    /// ```
     pub fn mintable(mut self, rule: AccessRule) -> Self {
         self.access_rules = self.access_rules.mintable(rule);
         self
     }
 
     /// Sets up who can burn (destroy) tokens of the resource
+    ///
+    /// Allows you to pass an [`AccessRule`] that defines who can burn tokens of the resource.
+    /// By default, burning is disabled for all users.
+    ///
+    /// #Examples
+    ///
+    /// ```rust, ignore
+    /// use tari_template_lib::auth::AccessRule;
+    /// use tari_template_lib::resource::builder::ResourceBuilder;
+    /// ResourceBuilder::confidential()
+    ///     .burnable(rule!(allow_all))
+    ///     .build();
+    /// ```
     pub fn burnable(mut self, rule: AccessRule) -> Self {
         self.access_rules = self.access_rules.burnable(rule);
         self
     }
 
     /// Sets up who can recall tokens of the resource.
-    /// A recall is the forceful withdrawal of tokens from any external vault
+    ///
+    /// A recall is the forceful withdrawal of **ALL** tokens from any external vault.
+    ///
+    /// Allows you to pass an [`AccessRule`] that defines who can recall tokens from a vault.
+    ///
+    /// By default, recalling is disabled for all users.
+    ///
+    /// # Examples
+    ///
+    /// ```rust, ignore
+    /// use tari_template_lib::auth::AccessRule;
+    /// use tari_template_lib::resource::builder::ResourceBuilder;
+    /// ResourceBuilder::confidential()
+    ///    .recallable(rule!(allow_all))
```

Comment:
Just cos, allow_all would be crazy lol
```suggestion
    ///    .recallable(rule!(require(resource_admin_badge)))
```